sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/PropertyConfigurator.java

```diff
@@ -0,0 +1,144 @@
+package com.microsoft.appcenter.analytics;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+import android.util.Property;
+
+import com.microsoft.appcenter.analytics.ingestion.models.one.CommonSchemaEventLog;
+import com.microsoft.appcenter.analytics.ingestion.models.one.json.CommonSchemaEventLogFactory;
+import com.microsoft.appcenter.channel.AbstractChannelListener;
+import com.microsoft.appcenter.channel.Channel;
+import com.microsoft.appcenter.channel.OneCollectorChannelListener;
+import com.microsoft.appcenter.ingestion.models.CustomPropertiesLog;
+import com.microsoft.appcenter.ingestion.models.Log;
+import com.microsoft.appcenter.ingestion.models.StartServiceLog;
+import com.microsoft.appcenter.ingestion.models.json.CustomPropertiesLogFactory;
+import com.microsoft.appcenter.ingestion.models.json.DefaultLogSerializer;
+import com.microsoft.appcenter.ingestion.models.json.LogSerializer;
+import com.microsoft.appcenter.ingestion.models.json.StartServiceLogFactory;
+import com.microsoft.appcenter.ingestion.models.one.AppExtension;
+import com.microsoft.appcenter.ingestion.models.one.CommonSchemaLog;
+import com.microsoft.appcenter.ingestion.models.one.SdkExtension;
+import com.microsoft.appcenter.utils.AppCenterLog;
+import com.microsoft.appcenter.utils.UUIDUtils;
+
+import java.util.Collection;
+
+import static com.microsoft.appcenter.utils.AppCenterLog.LOG_TAG;
+
+/**
+ * Allow overriding Part A properties.
+ */
+public class PropertyConfigurator extends AbstractChannelListener {
+
+    private String mAppName;
+
+    private String mAppVersion;
+
+    private String mAppLocale;
+
+    private AnalyticsTransmissionTarget mTransmissionTarget;
+
+    PropertyConfigurator(Channel channel, AnalyticsTransmissionTarget transmissionTarget) {
+        mTransmissionTarget = transmissionTarget;
+        if (channel != null) {
+            channel.addListener(this);
+        }
+    }
+
+    /**
+     *
```

Comment:
nit - remove the empty line; add description to `@param`.

Reply:
I had meant to remove this.  It's not listed in the One Collector listener so I'll follow that pattern.

Reply:
We usually never override Javadoc (`@Override` methods) unless the meaning actually changes. by default Javadoc is inherited.